packages/web-components/src/components/masthead/masthead-l1.ts

```diff
@@ -26,11 +42,233 @@ const { stablePrefix: ddsPrefix } = ddsSettings;
  */
 @customElement(`${ddsPrefix}-masthead-l1`)
 class DDSMastheadL1 extends StableSelectorMixin(LitElement) {
+  @property()
+  l1Data?: MastheadL1;
+
+  @property()
+  isMobileVersion?: boolean;
+
+  /**
+   * Renders L1 for desktop screensizes
+   *
+   * @returns {_TemplateResult} L1 for desktop screensizes
+   */
+  protected _renderL1TopNav() {
+    const { l1Data } = this;
+    const { url, title } = l1Data!;
+
+    return html`
+      ${!title
+        ? undefined
+        : html`
+            <dds-masthead-l1-name title="${title}" url="${ifDefined(url)}">
+            </dds-masthead-l1-name>
+          `}
+      <dds-top-nav-l1> L1 TOP NAV. </dds-top-nav-l1>
+    `;
+  }
+
+  /**
+   * Renders L1 for mobile screensizes.
+   *
+   * @returns {_TemplateResult} L1 for mobile screensizes.
+   */
+  protected _renderL1MobileNav() {
+    const { l1Data } = this;
+    const { url, title, actions, menuItems } = l1Data ?? {};
+    const { cta, login } = actions ?? {};
+
+    const { _toggleMobileSubsection: toggleMobileSubsection } = this
+      .constructor as typeof DDSMastheadL1;
+
+    setTimeout(() => {
+      console.clear();
+      console.log(l1Data);
+    }, 250);
+
+    return html`
+      <button
+        class="${prefix}--masthead__l1-title"
+        @click=${toggleMobileSubsection}>
+        ${title}${ChevronDown16()}
+      </button>
+      <ul class="${prefix}--masthead__l1-dropdown" hidden>
+        ${url
+          ? html`<li>
+              <a class="bx--masthead__l1-dropdown-item" href="${url}"
+                >Overview</a
+              >
+            </li>`
+          : ''}
+        ${menuItems?.map((menuItem) => this._renderL1MobileSubnav(menuItem))}
+        ${login
+          ? html`<li>
+              <a
+                class="bx--masthead__l1-dropdown-item bx--masthead__l1-dropdown-login"
+                href="${ifDefined(login.url)}"
+                >${login.title}${ArrowRight16()}</a
+              >
+            </li>`
+          : ''}
+        ${cta
+          ? html`<li>
+              <a
+                class="bx--masthead__l1-dropdown-item bx--masthead__l1-dropdown-cta"
+                href="${ifDefined(cta.url)}"
+                >${cta.title}${ArrowRight16()}</a
+              >
+            </li>`
+          : ''}
```

Comment:
Perhaps we should consider suppressing these if no URL is provided. Otherwise we'll get links that don't do anything.

Reply:
I added the ifDefined to satisfy the TS compiler. I figure the ternaries on lines 105 and 114 should be enough?

Reply:
I'm thinking about a scenario where, for example, `login` is passed in without a `url` property. As written, the code would render an `<a>` without a `href` attribute.